webknossos/webknossos/cli/download.py

```diff
@@ -72,15 +74,28 @@ def main(
 
     layers = layer if layer else None
     mags = mag if mag else None
+    url = resolve_short_link(url)
 
     with webknossos_context(token=token):
-        try:
-            Dataset.download(
-                dataset_name_or_url=url,
-                path=target,
-                bbox=bbox,
-                layers=layers,
-                mags=mags,
-            )
-        except AssertionError:
-            Annotation.download(annotation_id_or_url=url).save(target)
+        if re.match(_DATASET_URL_REGEX, url):
+            try:
+                Dataset.download(
+                    dataset_name_or_url=url,
+                    path=target,
+                    bbox=bbox,
+                    layers=layers,
+                    mags=mags,
+                )
+            except RuntimeError as err:
+                print(f"Dataset could not be downloaded: {err}")
+        elif re.match(_ANNOTATION_URL_REGEX, url):
+            try:
+                Annotation.download(annotation_id_or_url=url).save(target)
+            except AssertionError as err:
+                print(f"Annotation could not be downloaded: {err}")
```

Comment:
Do the other cli commands also implement this catch-print pattern? Is there a specific reason we don’t want to show the full stacktrace?

Also, why specifically catch AssertionError here, and RuntimeError above, but not all Exceptions or Errors?

I think I’d vote to bubble up the exceptions, that way the exit-code will also be 1 if the command was not successful. Consequently, the elese-branch could also raise instead of print. What do you think?

Reply:
I agree. I removed the try/catch and raised an error in the else case. It is useful when the exit code is 1 in case the command fails. Whether we show the stack trace, or not, depends somehow on the users. For debugging purposes, it is easier to have a stack trace, but it is also a highly technical error message that might confuse some users.